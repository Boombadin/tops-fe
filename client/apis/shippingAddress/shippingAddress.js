import axios from 'axios'
import { baseUrl } from '../config'

const setShippingLocation = (regionId, districtId, subdistrictId, zipcode, customerId) => {
  const options = {
    method: 'POST',
    url: `${baseUrl}/shipping/current`,
    data: {
      regionId: regionId,
      districtId: districtId,
      subdistrictId: subdistrictId,
      zipcode: zipcode,
      customerId: customerId
    }
  }
  
  return axios(options)
    .then(response => response.data)
    .catch(e => {
      throw e
    })
}

export default {
  setShippingLocation
}