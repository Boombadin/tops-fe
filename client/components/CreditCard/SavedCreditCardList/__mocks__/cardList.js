export default [
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: null,
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: null,
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '590925219668099073',
    isDefault: true,
    isExpired: false,
    maskedNumber: '411111XXXXXX1111',
    maskedNumberByStar: '**** **** **** 1111',
    type: 'Visa',
    typeIcon: '/icons/types-credit-card/visa-no-border.svg',
  },
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: '10',
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: 'KCC',
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '604578118707118081',
    isDefault: false,
    isExpired: false,
    maskedNumber: '512345XXXXXX2346',
    maskedNumberByStar: '**** **** **** 2346',
    type: 'MasterCard',
    typeIcon: '/icons/types-credit-card/master-no-border.svg',
  },
];
