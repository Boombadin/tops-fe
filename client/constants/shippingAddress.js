export const requiredFields = [
  'firstname',
  'lastname',
  'telephone',
  'house_no',
  'region_id',
  'district_id',
  'subdistrict_id',
  'postcode'
];
