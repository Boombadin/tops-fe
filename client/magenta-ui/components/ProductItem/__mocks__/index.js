export * from '@client/magenta-ui/components/ProductItem/__mocks__/product';
