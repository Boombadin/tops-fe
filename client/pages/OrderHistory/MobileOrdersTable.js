import { format } from 'date-fns';
import addHours from 'date-fns/add_hours';
import { isEmpty } from 'lodash';
import React, { Fragment } from 'react';
import InfiniteScroll from 'react-infinite-scroller';
import styled from 'styled-components';

import TrackOrderButton from '@/client/features/orderHistory/components/TrackOrderButton';
import { validURL } from '@/client/utils/order';
import withLocales from '@client/hoc/withLocales';
import { Loader } from '@client/magenta-ui';
import { formatPrice } from '@client/utils/price';

import { OrderAddress } from './elements';
import ReorderButton from './ReorderButton';

const ButtonFooterWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 32px;
`;

const timezoneBangkok = 7;
const MobileOrdersTable = ({
  orders,
  translate,
  lang,
  loading,
  onLoadmore,
  hasMoreItems,
  loadMore,
  disabled,
  handleOpenTrackClick,
}) => {
  return isEmpty(orders) ? null : (
    <Fragment>
      {loadMore ? (
        <InfiniteScroll
          className="product-grid--infinite"
          element="div"
          pageStart={1}
          hasMore={hasMoreItems}
          loader={null}
          threshold={10}
          loadMore={onLoadmore}
        >
          {orders.map(order => (
            <div className="mobile-table-row">
              <div className="info-row">
                <span className="title">
                  {translate('order_history.table.number')}
                </span>
                <span className="number">
                  <a
                    href={`/${lang?.url || 'th'}/order-detail/${
                      order?.entity_id
                    }`}
                  >
                    {order?.increment_id}
                  </a>
                </span>
              </div>
              <div className="info-row">
                <span className="title">
                  {translate('order_history.table.date')}
                </span>
                <span className="date-time">
                  <span className="date">
                    {format(
                      addHours(order.created_at, timezoneBangkok),
                      'DD/MM/YY',
                    )}
                  </span>
                  <span className="time">
                    {format(
                      addHours(order.created_at, timezoneBangkok),
                      'HH:mm',
                    )}
                  </span>
                </span>
              </div>
              <div className="info-row">
                <span className="title">
                  {translate('order_history.table.status')}
                </span>
                {order?.extension_attributes?.order_status ? (
                  <span
                    className={`status ${order?.extension_attributes?.order_status}`}
                  >
                    {order?.extension_attributes?.order_status}
                  </span>
                ) : (
                  <span className={`status new`}>
                    {translate('order_history.status.new')}
                  </span>
                )}
              </div>
              <div className="info-row">
                <span className="title">
                  {translate('order_history.table.price')}
                </span>
                <span className="price">
                  {formatPrice(order?.base_grand_total)}
                </span>
              </div>
              <div className="info-row">
                <span className="title">
                  {translate('order_history.table.address')}
                </span>
                <OrderAddress
                  translate={translate}
                  {...order.extension_attributes}
                />
              </div>
              <ButtonFooterWrapper>
                {order?.extension_attributes?.tracking_info?.length > 0 &&
                  validURL(
                    order?.extension_attributes?.tracking_info[0]?.track_link,
                  ) && (
                    <TrackOrderButton
                      orderId={order?.entity_id}
                      handleTrackClick={() =>
                        handleOpenTrackClick(
                          order?.extension_attributes?.tracking_info[0]
                            ?.track_link,
                        )
                      }
                    />
                  )}
                <ReorderButton orderId={order.entity_id} disabled={disabled} />
              </ButtonFooterWrapper>
            </div>
          ))}
          {loading && <Loader className="load-more" active inline="centered" />}
        </InfiniteScroll>
      ) : (
        ''
      )}
    </Fragment>
  );
};

export default withLocales(MobileOrdersTable);
