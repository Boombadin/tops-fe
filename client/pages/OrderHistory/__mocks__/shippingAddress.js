export const shippingAddress = [
  {
    shipping: {
      address: {
        address_type: 'shipping',
        city: 'กรุงเทพมหานคร',
        company: 'N/A',
        country_id: 'TH',
        email: 'soulcietycomsci+2@gmail.com',
        entity_id: 1243575,
        extension_attributes: {
          custom_attributes: [
            { attribute_code: 'moo', value: '1' },
            { attribute_code: 'soi', value: '1' },
            { attribute_code: 'address_name', value: 'name' },
            { attribute_code: 'house_no', value: '111' },
            { attribute_code: 'village_name', value: 'village_name' },
            { attribute_code: 'road', value: '12' },
            { attribute_code: 'district', value: 'บางรัก' },
            { attribute_code: 'subdistrict', value: 'บางรัก' },
          ],
        },
        firstname: 'test',
        lastname: 'test1',
        parent_id: 622300,
        postcode: '10500',
        region: 'กรุงเทพมหานคร',
        region_code: 'bkk',
        region_id: 534,
        street: ['N/A'],
        telephone: '0876612321',
      },
      method: 'tops_express',
    },
  },
];
