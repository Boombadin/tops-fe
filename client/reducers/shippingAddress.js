import { find, get as prop } from 'lodash'
import ShippingAddressApi from '../apis/shippingAddress'
import RegionApi from '../apis/region'
import { getDefaultShippingAddressSelector } from '../selectors'

export const TYPES = {
  SET_CURRENT_PROVINCE: 'SET_CURRENT_PROVINCE',
  SET_CURRENT_DISTRICT: 'SET_CURRENT_DISTRICT',
  SET_CURRENT_SUB_DISTRICT: 'SET_CURRENT_SUB_DISTRICT',
  SET_CURRENT_ZIPCODE: 'SET_CURRENT_ZIPCODE'
}

export const saveShippingLocation = (regionId, districtId, subdistrictId, zipcode) => {
  return async (dispatch, getState) => {
    const { customer } = getState()
    const response = await ShippingAddressApi.setShippingLocation(regionId, districtId, subdistrictId, zipcode, prop(customer, 'items.id', 0))
    return response
  }
}

export const setCurrentShippingLocation = (regionId, districtId, subdistrictId, zipcode) => {
  return (dispatch) => {
    dispatch(setCurrentProvince(regionId))
    dispatch(setCurrentDistrict(districtId))
    dispatch(setCurrentSubDistrict(subdistrictId))
    dispatch(setZipcode(zipcode))
  }
}

export const setCurrentProvince = regionId => ({
  type: TYPES.SET_CURRENT_PROVINCE,
  payload: { regionId }
})

export const setCurrentDistrict = districtId => ({
  type: TYPES.SET_CURRENT_DISTRICT,
  payload: { districtId }
})

export const setCurrentSubDistrict = subdistrictId => ({
  type: TYPES.SET_CURRENT_SUB_DISTRICT,
  payload: { subdistrictId }
})

export const setZipcode = zipcode => ({
  type: TYPES.SET_CURRENT_ZIPCODE,
  payload: { zipcode }
})

const initialState = {
  province: '',
  district: '',
  subdistrict: '',
  zipcode: ''
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.SET_CURRENT_PROVINCE: {
      const { regionId } = action.payload
      return { ...state, ...{ province: regionId } } 
    }

    case TYPES.SET_CURRENT_DISTRICT: {
      const { districtId } = action.payload
      return { ...state, ...{ district: districtId } } 
    }

    case TYPES.SET_CURRENT_SUB_DISTRICT: {
      const { subdistrictId } = action.payload
      return { ...state, ...{ subdistrict: subdistrictId } } 
    }

    case TYPES.SET_CURRENT_ZIPCODE: {
      const { zipcode } = action.payload
      return { ...state, ...{ zipcode: zipcode } } 
    }

    default:
      return state
  }
}

export default reducer
