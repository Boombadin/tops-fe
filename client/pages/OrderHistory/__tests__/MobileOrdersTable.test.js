import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { shippingAddress } from '../__mocks__/shippingAddress';
import MobileOrdersTable from '../MobileOrdersTable';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});

const TEST_ID = {
  BUTTON_TRACK: 'btn-viewTrackOrderButton-handleTrackClick-621636',
};

describe('features/orderDetail/components/MobileOrdersTable', () => {
  const props = {
    orders: [
      {
        entity_id: 621636,
        increment_id: 'FC432200605044',
        created_at: '2020-06-05 03:30:32',
        extension_attributes: {
          order_status: 'คำสั่งซื้อสมบูรณ์',
          shipping_assignments: shippingAddress,
          tracking_info: [
            {
              order_id: '621636',
              track_link: 'https://www.google.co.th/',
              track_number: 'NOT AVAILABLE',
            },
          ],
        },
        base_grand_total: 1692,
      },
    ],
    onLoadmore: () => {},
    disabled: false,
    handleOpenTrackClick: jest.fn(),
    hasMoreItems: true,
    loadMore: true,
    loading: false,
  };

  test('it should be match snapshot', () => {
    const { asFragment } = render(<MobileOrdersTable {...props} />);
    expect(asFragment(<MobileOrdersTable {...props} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when order empty', () => {
    const customProps = {
      ...props,
      orders: null,
    };
    const { asFragment } = render(<MobileOrdersTable {...customProps} />);
    expect(asFragment(<MobileOrdersTable {...props} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when order status null or empty', () => {
    const customProps = {
      ...props,
      orders: [
        {
          ...props?.orders[0],
          extension_attributes: {
            ...props?.orders[0].extension_attributes,
            order_status: null,
          },
        },
      ],
    };
    const { asFragment } = render(<MobileOrdersTable {...customProps} />);
    expect(asFragment(<MobileOrdersTable {...props} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when loadMore false', () => {
    const customProps = {
      ...props,
      loadMore: false,
    };
    const { asFragment } = render(<MobileOrdersTable {...customProps} />);
    expect(asFragment(<MobileOrdersTable {...props} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when loading', () => {
    const customProps = {
      ...props,
      loading: true,
    };
    const { asFragment } = render(<MobileOrdersTable {...customProps} />);
    expect(
      asFragment(<MobileOrdersTable {...customProps} />),
    ).toMatchSnapshot();
  });

  test('hidden button tracking when track_link null | it should be match snapshot', () => {
    const customProps = {
      ...props,
      orders: [
        {
          ...props?.orders[0],
          extension_attributes: {
            ...props?.orders[0].extension_attributes,
            tracking_info: [
              {
                ...props?.orders[0].extension_attributes?.tracking_info[0],
                track_link: null,
              },
            ],
          },
        },
      ],
    };

    const { asFragment } = render(<MobileOrdersTable {...customProps} />);
    expect(
      asFragment(<MobileOrdersTable {...customProps} />),
    ).toMatchSnapshot();
  });

  test('click open tracking from web grap | it should be match snapshot', () => {
    const { getByTestId, asFragment } = render(
      <MobileOrdersTable {...props} />,
    );
    const buttonViewOrderTrack = getByTestId(TEST_ID.BUTTON_TRACK);
    expect(buttonViewOrderTrack).toBeInTheDocument();
    fireEvent.click(buttonViewOrderTrack);
    expect(asFragment()).toMatchSnapshot();
  });
});
