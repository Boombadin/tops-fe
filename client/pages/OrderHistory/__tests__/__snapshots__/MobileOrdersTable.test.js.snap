// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/orderDetail/components/MobileOrdersTable click open tracking from web grap | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-grid--infinite"
  >
    <div
      class="mobile-table-row"
    >
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.number
        </span>
        <span
          class="number"
        >
          <a
            href="/th/order-detail/621636"
          >
            FC432200605044
          </a>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.date
        </span>
        <span
          class="date-time"
        >
          <span
            class="date"
          >
            05/06/20
          </span>
          <span
            class="time"
          >
            10:30
          </span>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.status
        </span>
        <span
          class="status คำสั่งซื้อสมบูรณ์"
        >
          คำสั่งซื้อสมบูรณ์
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.price
        </span>
        <span
          class="price"
        >
          1,692.00
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.address
        </span>
        <span
          class="address"
        >
          shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
        </span>
      </div>
      <div
        class="MobileOrdersTable__ButtonFooterWrapper-sc-1em5rrq-0 lcLrRM"
      >
        <button
          class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
          color="#544943"
          data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
          radius="5"
        >
          order_history.table.track_order
        </button>
        <button
          class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
          color="#544943"
          data-testid="btn-addReorder-reorder-621636"
          radius="5"
        >
          order_history.table.repeat
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/MobileOrdersTable hidden button tracking when track_link null | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-grid--infinite"
  >
    <div
      class="mobile-table-row"
    >
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.number
        </span>
        <span
          class="number"
        >
          <a
            href="/th/order-detail/621636"
          >
            FC432200605044
          </a>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.date
        </span>
        <span
          class="date-time"
        >
          <span
            class="date"
          >
            05/06/20
          </span>
          <span
            class="time"
          >
            10:30
          </span>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.status
        </span>
        <span
          class="status คำสั่งซื้อสมบูรณ์"
        >
          คำสั่งซื้อสมบูรณ์
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.price
        </span>
        <span
          class="price"
        >
          1,692.00
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.address
        </span>
        <span
          class="address"
        >
          shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
        </span>
      </div>
      <div
        class="MobileOrdersTable__ButtonFooterWrapper-sc-1em5rrq-0 lcLrRM"
      >
        <button
          class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
          color="#544943"
          data-testid="btn-addReorder-reorder-621636"
          radius="5"
        >
          order_history.table.repeat
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/MobileOrdersTable it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-grid--infinite"
  >
    <div
      class="mobile-table-row"
    >
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.number
        </span>
        <span
          class="number"
        >
          <a
            href="/th/order-detail/621636"
          >
            FC432200605044
          </a>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.date
        </span>
        <span
          class="date-time"
        >
          <span
            class="date"
          >
            05/06/20
          </span>
          <span
            class="time"
          >
            10:30
          </span>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.status
        </span>
        <span
          class="status คำสั่งซื้อสมบูรณ์"
        >
          คำสั่งซื้อสมบูรณ์
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.price
        </span>
        <span
          class="price"
        >
          1,692.00
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.address
        </span>
        <span
          class="address"
        >
          shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
        </span>
      </div>
      <div
        class="MobileOrdersTable__ButtonFooterWrapper-sc-1em5rrq-0 lcLrRM"
      >
        <button
          class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
          color="#544943"
          data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
          radius="5"
        >
          order_history.table.track_order
        </button>
        <button
          class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
          color="#544943"
          data-testid="btn-addReorder-reorder-621636"
          radius="5"
        >
          order_history.table.repeat
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/MobileOrdersTable it should be match snapshot when loadMore false 1`] = `<DocumentFragment />`;

exports[`features/orderDetail/components/MobileOrdersTable it should be match snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="product-grid--infinite"
  >
    <div
      class="mobile-table-row"
    >
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.number
        </span>
        <span
          class="number"
        >
          <a
            href="/th/order-detail/621636"
          >
            FC432200605044
          </a>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.date
        </span>
        <span
          class="date-time"
        >
          <span
            class="date"
          >
            05/06/20
          </span>
          <span
            class="time"
          >
            10:30
          </span>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.status
        </span>
        <span
          class="status คำสั่งซื้อสมบูรณ์"
        >
          คำสั่งซื้อสมบูรณ์
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.price
        </span>
        <span
          class="price"
        >
          1,692.00
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.address
        </span>
        <span
          class="address"
        >
          shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
        </span>
      </div>
      <div
        class="MobileOrdersTable__ButtonFooterWrapper-sc-1em5rrq-0 lcLrRM"
      >
        <button
          class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
          color="#544943"
          data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
          radius="5"
        >
          order_history.table.track_order
        </button>
        <button
          class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
          color="#544943"
          data-testid="btn-addReorder-reorder-621636"
          radius="5"
        >
          order_history.table.repeat
        </button>
      </div>
    </div>
    <div
      class="ui active centered inline loader load-more"
    />
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/MobileOrdersTable it should be match snapshot when order empty 1`] = `<DocumentFragment />`;

exports[`features/orderDetail/components/MobileOrdersTable it should be match snapshot when order status null or empty 1`] = `
<DocumentFragment>
  <div
    class="product-grid--infinite"
  >
    <div
      class="mobile-table-row"
    >
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.number
        </span>
        <span
          class="number"
        >
          <a
            href="/th/order-detail/621636"
          >
            FC432200605044
          </a>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.date
        </span>
        <span
          class="date-time"
        >
          <span
            class="date"
          >
            05/06/20
          </span>
          <span
            class="time"
          >
            10:30
          </span>
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.status
        </span>
        <span
          class="status new"
        >
          order_history.status.new
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.price
        </span>
        <span
          class="price"
        >
          1,692.00
        </span>
      </div>
      <div
        class="info-row"
      >
        <span
          class="title"
        >
          order_history.table.address
        </span>
        <span
          class="address"
        >
          shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
        </span>
      </div>
      <div
        class="MobileOrdersTable__ButtonFooterWrapper-sc-1em5rrq-0 lcLrRM"
      >
        <button
          class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
          color="#544943"
          data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
          radius="5"
        >
          order_history.table.track_order
        </button>
        <button
          class="ReorderButton__Reorder-sc-1een2x9-0 liOase"
          color="#544943"
          data-testid="btn-addReorder-reorder-621636"
          radius="5"
        >
          order_history.table.repeat
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
